app/src/main/kotlin/com/intive/picover/images/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.intive.picover.images.viewmodel
+
+import android.net.Uri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.images.repository.ImagesRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<LatestFirebaseStorageState>(LatestFirebaseStorageState.Loading)
+	val uiState: StateFlow<LatestFirebaseStorageState> = _uiState
+
+	init {
+		viewModelScope.launch(Dispatchers.IO) {
+			fetchPics().collect()
+		}
+	}
+
+	private suspend fun fetchPics(): Flow<List<Uri>> = callbackFlow<List<Uri>> {
+		_uiState.value = LatestFirebaseStorageState.Loading
+		try {
+			val uris = repository.fetchItems().map { repository.downloadUri(it) }
+			_uiState.value = LatestFirebaseStorageState.Success(uris.sorted())
+			trySend(uris)
+		} catch (e: Exception) {
+			_uiState.value = LatestFirebaseStorageState.Error(e)
+		}
+		awaitClose { close() }
+	}
+
+	sealed class LatestFirebaseStorageState {
```

Comment:
minor: This class could be in another file, in the future holding everything in one place could make a mess IMO.

Reply:
We can discuss it during the Guild.

Reply:
We have agreed that it needs to go to a separate file.